pkg/apis/topology/v1alpha2/helper/attribute/attribute.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package attribute
+
+import (
+	"github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+)
+
+// Find looks for attribute by its name in the provided AtributeList.
+// If found, returns the value of the attribute and an explicit boolean value equals true
+// Otherwise, returns the zero value and explicit boolean value equals false
+// In case there are duplicates in the attribute list, returns the first attribute from the left
+// found in iteration order, iterating on the attribute list treating it as a golang slice.
+func Find(attributes v1alpha2.AttributeList, name string) (string, bool) {
+	attr := findByName(attributes, name)
+	if attr == nil {
+		return "", false
+	}
+	return attr.Value, true
+}
+
+// Update updates or adds an attribute to the provided AttributeList, preserving the existing order.
+// The attribute name must be non-zero, the value can be zero.
+// If the attribute is already present in the AttributeList, its value is overridden with the given one.
+// If the attribute was not present in the AttributeList, adds a new attribute minimizing the changes
+// to the attribute list.
+// Returns the updated AttributeList and true if an attribute was created, false if modified in place.
+func Update(attributes v1alpha2.AttributeList, name, value string) (v1alpha2.AttributeList, bool) {
+	if name == "" {
```

Comment:
should we also check if `value == ""` ?

Reply:
I think empty values are fine. We have examples in k8s world for empty labels and attributes which are totally legit and largely used. Empty Name (=key) OTOH is awkward